acceptance/cluster/localcluster.go

```diff
@@ -504,11 +504,18 @@ func (l *LocalCluster) Start() {
 func (l *LocalCluster) Assert(t *testing.T) {
 	const almostZero = 50 * time.Millisecond
 	filter := func(ch chan Event, wait time.Duration) *Event {
+		waitTimer := time.NewTimer(wait)
+		waitTimerRead := false
 		for {
+			if !waitTimer.Reset(wait) && !waitTimerRead {
+				<-waitTimer.C
+			}
+			waitTimerRead = false
 			select {
 			case act := <-ch:
 				return &act
-			case <-time.After(wait):
+			case <-waitTimer.C:
+				waitTimerRead = true
```

Comment:
I think we should also be calling `time.Timer.Stop` when the loop exits. Failure to do that prevents the timer memory from being reclaimed until the timer expires.